openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -169,43 +204,150 @@ def run(self, problem):
+                    full_conns = full_param_conns.get(param)
+                    if full_conns.intersection(rels):
+                        continue
+
+                    rel_idx = set()
+                    for target, idx in iteritems(sub_conns):
+
+                        # If a target of the indexed desvar connection is
+                        # in the relevant path for this constraint, then
+                        # those indices are relevant.
+                        if target in rels:
+                            rel_idx.update(idx)
+
+                    nrel = len(rel_idx)
+                    if nrel > 0:
+
+                        if jac is None:
+                            jac = {}
+
+                        if param not in jac:
+                            # A coo matrix for the Jacobian
+                            # mat = {'coo':[row, col, data],
+                            #        'shape':[nrow, ncols]}
+                            coo = {}
+                            coo['shape'] = [size, len(param_vals[param])]
+                            jac[param] = coo
+
+                        row = []
+                        col = []
+                        for i in range(size):
+                            row.extend([i]*nrel)
+                            col.extend(rel_idx)
+                        data = np.ones((len(row), ))
+
+                        jac[param]['coo'] = [np.array(row), np.array(col), data]
+
+                        if name not in self.sub_sparsity:
+                            self.sub_sparsity[name] = {}
+                        self.sub_sparsity[name][param] = np.array(list(rel_idx))
+
+
                 opt_prob.addConGroup(name, size, lower=lower, upper=upper,
-                                     wrt=wrt)
+                                     wrt=wrt, jac=jac)
 
         # Add all inequality constraints
         incons = self.get_constraints(ctype='ineq', lintype='nonlinear')
         self.quantities += list(incons)
 
         for name in self.get_constraints(ctype='ineq'):
-            size = con_meta[name]['size']
+            meta = con_meta[name]
+            size = meta['size']
 
             # Bounds - double sided is supported
-            lower = con_meta[name]['lower']
-            upper = con_meta[name]['upper']
+            lower = meta['lower']
+            upper = meta['upper']
 
             # Sparsify Jacobian via relevance
-            wrt = rel.relevant[name].intersection(indep_list)
+            rels = rel.relevant[name]
+            wrt = rels.intersection(indep_list)
             self.sparsity[name] = wrt
 
-            if con_meta[name]['linear'] is True:
+            if meta['linear'] is True:
                 opt_prob.addConGroup(name, size, upper=upper, lower=lower,
                                      linear=True, wrt=wrt,
                                      jac=self.lin_jacs[name])
             else:
+
+                jac = None
```

Comment:
This whole block is largely duplicated from the eq constraints isn't it? Move into a helper function?